datafusion/expr-common/src/type_coercion/binary.rs

```diff
@@ -869,54 +867,6 @@ fn get_wider_decimal_type(
     }
 }
 
-/// Returns the wider type among arguments `lhs` and `rhs`.
-/// The wider type is the type that can safely represent values from both types
-/// without information loss. Returns an Error if types are incompatible.
-pub fn get_wider_type(lhs: &DataType, rhs: &DataType) -> Result<DataType> {
```

Comment:
this was used in `array_concat` only to compute the return type, and `array_concat` errors at runtime anyways if you pass it arrays with different types.

Thus this code is unecessary

```
> select array_concat([arrow_cast('1', 'LargeUtf8')], ['2']);
Arrow error: Invalid argument error: It is not possible to concatenate arrays of different data types.
```